kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js

```diff
@@ -488,83 +494,102 @@ define([
-                            }
+        async function render() {
+            const inputControl = makeInputControl(),
+                content = div({ class: 'input-group', style: { width: '100%' } }, inputControl);
+
+            ui.setContent('input-container', content);
+            const dropdown = $(ui.getElement('input-container.input'));
+            const data = [];
+
+            if (config.isViewOnly) {
+                let viewData = {
+                    selected: true,
+                    id: config.initialValue || undefined,
+                    text: config.initialValue,
+                };
+                if (config.initialDisplayValue) {
+                    if (typeof config.initialDisplayValue === 'string') {
+                        viewData.text = config.initialDisplayValue;
+                    } else if (
+                        config.initialDisplayValue !== null &&
+                        typeof config.initialDisplayValue === 'object'
+                    ) {
+                        viewData = Object.assign(viewData, config.initialDisplayValue);
+                        if (!viewData.text) {
+                            viewData.text = viewData.id;
                         }
                     }
-                    data.push(viewData);
                 }
-                dropdown.select2({
-                    allowClear: true,
-                    disabled: config.isViewOnly,
-                    templateResult: formatObjectDisplay,
-                    templateSelection: selectionTemplate,
-                    data: config.isViewOnly ? data : null,
-                    ajax: {
-                        delay: 250,
-                        processResults: (data) => {
-                            // update the currently selected one if applicable
-                            const currentData = dropdown.select2('data')[0];
-                            if (currentData) {
-                                const updatedData = data.results.filter((item) => {
-                                    return item.id === currentData.id;
-                                });
-                                if (updatedData.length) {
-                                    for (const key of Object.keys(updatedData[0])) {
-                                        currentData[key] = updatedData[0][key];
-                                    }
-                                    dropdown.trigger('change');
+                data.push(viewData);
+            } else if (model.value && !model.displayValue && ddOptions.exact_match_on) {
+                const searchResults = await fetchData(model.value);
+                // verify that we have an exact match
+                let match = null;
+                for (const result of searchResults) {
+                    if (
+                        model.value.toLowerCase() === result[ddOptions.exact_match_on].toLowerCase()
+                    ) {
```

Comment:
do we need to do any of that tedious stuff like trimming off whitespace or ensuring that there's some content in the match string, or is that done earlier on?

Reply:
Yes, certainly. Don't know how I missed that. I'm just making silly assumptions that the database call probably won't honor.